### ISSUE TITLE
[Bug] Node version error, cannot complete step 4.

### ISSUE THREAD

**Issue by gitgithan** [skills/deploy-to-azure#54]
![267984735-b2358aaf-9827-4ea6-9f9c-5b473d73c99a](https://github.com/skills/deploy-to-azure/assets/29853829/6390a9f2-356a-4d0c-af1c-6ce755d4c38a)

I used a workaround from https://github.com/skills/deploy-to-azure/issues/39#issuecomment-1682483257, by adding 
```
- uses: actions/setup-node@v3
       with: 
         node-version: 16
 ```
 after both checkout actions.
 
 However, since the step has already updated to 5, my new PR (with a commit adding node 16) will not get automatically squash merged by `run: gh pr merge --squash staging-test` because below check fails since now is step 5
 ```
     if: >-
      ${{ !github.event.repository.is_template
          && needs.get_current_step.outputs.current_step == 4 }}
 ```
 
 The following will not run too. Not sure why this script wants to Merge main into each branch, since throughout the exercises, after `run: gh pr merge --squash`, only `main` branch is meaningful and the others are left hanging useless. 
 ```
 run: |
          chmod +x ./.github/script/initialize-repository.sh
          ./.github/script/initialize-repository.sh
 ```
  
  
  Step 5 actions also cannot complete because of same issue

**Comment by alopdele** [comment]
I have the same error when I do step 4. I don't know how to fix it.

**Comment by gitgithan** [comment]
Fixed by #59

**Comment by gitgithan** [comment]
Fixed by #59
